base/doc/clsguide.tex

```diff
@@ -1567,312 +1058,255 @@ \subsection{Normalising spacing}
-   ...
-   \DeclareOldFontCommand{\sc}{\normalfont\scshape}{\mathsc}
+   \section{Stuff about sequences $\seq{x}$}
 \end{verbatim}
-This means, for example, that |{\bf\it text}| will produce medium
-weight (rather than bold) italic, thus: \textit{text}.
-
-\begin{decl}
-   |\normalsize| \\
-   |\@normalsize|
-\end{decl}
-The command |\@normalsize| is retained for compatibility with
-\LaTeX~2.09 packages which may have used its value; but redefining it
-in a class file will have no effect since it is always reset to have
-the same meaning as |\normalsize|.
 
-This means that classes \emph{must} now redefine |\normalsize| rather
-than redefining |\@normalsize|; for example (a rather incomplete one):
-\begin{verbatim}
-   \renewcommand{\normalsize}{\fontsize{10}{12}\selectfont}
-\end{verbatim}
-Note that |\normalsize| is defined by the \LaTeX{} kernel to be an
-error message.
+Note also that there is no need to put a |\relax| before the |\ifmmode| at the
+beginning of the definition; this is because the protection given by this
+|\relax| against expansion at the wrong time will be provided internally.
 
 \begin{decl}
-   |\tiny| |\footnotesize| |\small| |\large|\\
-   |\Large| |\LARGE| |\huge| |\Huge|
+  |\CheckCommand| \arg{cmd} \oarg{num} \oarg{default}
+     \arg{definition}\\
+  |\CheckCommand*| \arg{cmd} \oarg{num} \oarg{default}
+     \arg{definition}
 \end{decl}
-None of these other `standard' size-changing commands are defined in
-the kernel: each needs to be defined in a class file if you need it.
-They are all defined by the standard classes.
-
-This means you should use |\renewcommand| for |\normalsize| and
-|\newcommand| for the other size-changing commands.
+This takes the same arguments as |\newcommand| but, rather than define \m{cmd},
+it just checks that the current definition of \m{cmd} is exactly as given by
+\m{definition}. An error is raised if these definitions differ.
 
+This command is useful for checking the state of the system before your package
+starts altering the definitions of commands. It allows you to check, in
+particular, that no other package has redefined the same command.
 
-\subsection{Obsolete commands}
-
-Some packages will not work with \LaTeXe{}, normally because they relied
-on an internal \LaTeX{} command which was never supported and has now
-changed, or been removed.
-
-In many cases there will now be a robust, high-level means of
-achieving what previously required low-level commands.  Please consult
-Section~\ref{Sec:commands} to see if you can now use the \LaTeXe{}
-class and package writers commands.
-
-Also, of course, if your package or class redefined any of the kernel
-commands (i.e.~those defined in the files |latex.tex|, |slitex.tex|,
-|lfonts.tex|, |sfonts.tex|) then you will need to check it very
-carefully against the new kernel in case the implementation has
-changed or the command no longer exists in the \LaTeX2e{} kernel.
+\subsection{Option declaration}
+\label{Sec:commands.options.dec}
```

Comment:
A bit contradictory to me: if the "simple text" approach is considered deprecated (this is a subsection under section "_Deprecated commands in wide use_"), why they are still introduced in previous [section "_Declaring options_"](https://github.com/latex3/latex2e/pull/980/files#diff-41bff0166bc2d81d83fbc90b944f0ae0b22f3f7af905c00f1c17bae84382f471R421)?

> \subsection{Declaring options}
> [...]
> As the `classical' simple text approach is conceptually more
> straight-forward to illustrate, it is used here to show the general structure:
> see Section~\ref{Sec:opt:keyval} for full details of the key--value approach.

Or maybe a reference to current subsection can be added there, to prompt authors this approach are deprecated.

Reply:
Because it's simpler ;)

Seriously, keyvals are powerful bu they need a bit of thought. I didn't think I could just update the text here to do the 'in place' demo in the way the old version did.